3.6/debian-9/rootfs/opt/bitnami/scripts/libmongodb.sh

```diff
@@ -511,14 +672,37 @@ EOF
         )
     fi
 
-    if [[ -n "$MONGODB_USERNAME" ]] && [[ -n "$MONGODB_PASSWORD" ]] && [[ -n "$MONGODB_DATABASE" ]]; then
-        info "Creating '$MONGODB_USERNAME' user..."
+    if [[ -n "$MONGODB_USERNAME" ]]; then
+        local databases usernames passwords
 
-        result=$(
-            mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<EOF
-db.getSiblingDB('$MONGODB_DATABASE').createUser({ user: '$MONGODB_USERNAME', pwd: '$MONGODB_PASSWORD', roles: [{role: 'readWrite', db: '$MONGODB_DATABASE'}] })
-EOF
-        )
+        # Fill in arrays called databases, usernames and passwords with
+        # information from matching environment variables, there will only be
+        # one element at max in them.
+        mongodb_auth single
+        if [[ -n "$MONGODB_DATABASE" ]]; then
+            mongodb_create_user "${usernames[0]}" "${passwords[0]}" "${databases[0]}"
+        else
+            mongodb_create_user "${usernames[0]}" "${passwords[0]}"
+        fi
```

Comment:
You already checked that `MONGODB_USERNAME ` is set, and you have check in `mongodb_create_user` to check if the password or database name is set. Then I guess you can simply do this:
```suggestion
        mongodb_create_user "$MONGODB_USERNAME" "$MONGODB_PASSWORD" "$MONGODB_DATABASE"
```

Reply:
Good one!